### PR TITLE
Lower JVM heap size when low memory is detected

### PR DESCRIPTION
**Pull request by ekohl**
This is untested, but should help especially when testing and you don't really need performance but it just needs to be up.

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -411,15 +411,24 @@
   }
 
   $server_jvm_java_bin      = '/usr/bin/java'
-  $server_jvm_min_heap_size = '2G'
-  $server_jvm_max_heap_size = '2G'
   $server_jvm_extra_args    = '-XX:MaxPermSize=256m'
 
+  # This is some very trivial "tuning". See the puppet reference:
+  # https://docs.puppet.com/puppetserver/latest/tuning_guide.html
+  if $::memorysize_mb >= 2048 {
```
Comment (domcleal) [inline manifests/params.pp]: This is a string on pre-4.x versions, consider using `0 + $::memorysize_mb` to force the type change.

**Comment by ekohl** [comment]
I also noticed that VMs with 1G RAM and crashkernel enabled end up with 992MB RAM so fail to pass with 1G. I'm adding another case for that as well.

File: manifests/params.pp (hunk 1 above)
Comment (ekohl) [inline manifests/params.pp]: Fixed

**Comment by mmoll** [comment]
merged, bedankt @ekohl!

**Pull request merged**